File: icure-sdk/src/commonMain/kotlin/com/icure/sdk/model/embed/InvoicingCode.kt
package com.icure.sdk.model.embed

import kotlin.Boolean
import kotlin.Double
import kotlin.Int
import kotlin.Long
import kotlin.String
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
public sealed interface InvoicingCode : Encryptable {
  public val id: String?

  public val dateCode: Long?

  public val logicalId: String?

  public val label: String?

  public val userId: String?

  public val contactId: String?

  public val serviceId: String?

  public val tarificationId: String?

  public val code: String?

  public val paymentType: PaymentType?

  public val paid: Double?

  public val totalAmount: Double?

  public val reimbursement: Double?

  public val patientIntervention: Double?

  public val doctorSupplement: Double?

  public val conventionAmount: Double?

  public val vat: Double?

  public val error: String?

  public val contract: String?

  public val contractDate: Long?

  public val units: Int?

  public val side: Int?

  public val timeOfDay: Int?

  public val eidReadingHour: Int?

  public val eidReadingValue: String?

  public val override3rdPayerCode: Int?

  public val override3rdPayerReason: String?

  public val transplantationCode: Int?

  public val prescriberNorm: Int?

  public val percentNorm: Int?

  public val prescriberNihii: String?

  public val relatedCode: String?

  public val prescriptionDate: Long?

  public val derogationMaxNumber: Int?

  public val prescriberSsin: String?

  public val prescriberLastName: String?

  public val prescriberFirstName: String?

  public val prescriberCdHcParty: String?

  public val locationNihii: String?

  public val locationCdHcParty: String?

  public val locationService: Int?

  public val admissionDate: Long?

  public val canceled: Boolean?

  public val accepted: Boolean?

  public val pending: Boolean?

  public val resent: Boolean?

  public val archived: Boolean?

  public val lost: Boolean?

  public val insuranceJustification: Int?

  public val cancelPatientInterventionReason: Int?

  public val status: Long?

  override val encryptedSelf: String?
}

@SerialName
public data class DecryptedInvoicingCode(
  override val id: String? = null,
  override val dateCode: Long? = null,
  override val logicalId: String? = null,
  override val label: String? = null,
  override val userId: String? = null,
  override val contactId: String? = null,
  override val serviceId: String? = null,
  override val tarificationId: String? = null,
  override val code: String? = null,
  override val paymentType: PaymentType? = null,
  override val paid: Double? = null,
  override val totalAmount: Double? = null,
  override val reimbursement: Double? = null,
  override val patientIntervention: Double? = null,
  override val doctorSupplement: Double? = null,
  override val conventionAmount: Double? = null,
  override val vat: Double? = null,
  override val error: String? = null,
  override val contract: String? = null,
  override val contractDate: Long? = null,
  override val units: Int? = null,
  override val side: Int? = null,
  override val timeOfDay: Int? = null,
  override val eidReadingHour: Int? = null,
  override val eidReadingValue: String? = null,
  override val override3rdPayerCode: Int? = null,
  override val override3rdPayerReason: String? = null,
  override val transplantationCode: Int? = null,
  override val prescriberNorm: Int? = null,
  override val percentNorm: Int? = null,
  override val prescriberNihii: String? = null,
  override val relatedCode: String? = null,
  override val prescriptionDate: Long? = null,
  override val derogationMaxNumber: Int? = null,
  override val prescriberSsin: String? = null,
  override val prescriberLastName: String? = null,
  override val prescriberFirstName: String? = null,
  override val prescriberCdHcParty: String? = null,
  override val locationNihii: String? = null,
  override val locationCdHcParty: String? = null,
  override val locationService: Int? = null,
  override val admissionDate: Long? = null,
  override val canceled: Boolean? = null,
  override val accepted: Boolean? = null,
  override val pending: Boolean? = null,
  override val resent: Boolean? = null,
  override val archived: Boolean? = null,
  override val lost: Boolean? = null,
  override val insuranceJustification: Int? = null,
  override val cancelPatientInterventionReason: Int? = null,
  override val status: Long? = null,
  override val encryptedSelf: String? = null,
) : InvoicingCode

@SerialName
public data class EncryptedInvoicingCode(
  override val id: String? = null,
  override val dateCode: Long? = null,
  override val logicalId: String? = null,
  override val label: String? = null,
  override val userId: String? = null,
  override val contactId: String? = null,
  override val serviceId: String? = null,
  override val tarificationId: String? = null,
  override val code: String? = null,
  override val paymentType: PaymentType? = null,
  override val paid: Double? = null,
  override val totalAmount: Double? = null,
  override val reimbursement: Double? = null,
  override val patientIntervention: Double? = null,
  override val doctorSupplement: Double? = null,
  override val conventionAmount: Double? = null,
  override val vat: Double? = null,
  override val error: String? = null,
  override val contract: String? = null,
  override val contractDate: Long? = null,
  override val units: Int? = null,
  override val side: Int? = null,
  override val timeOfDay: Int? = null,
  override val eidReadingHour: Int? = null,
  override val eidReadingValue: String? = null,
  override val override3rdPayerCode: Int? = null,
  override val override3rdPayerReason: String? = null,
  override val transplantationCode: Int? = null,
  override val prescriberNorm: Int? = null,
  override val percentNorm: Int? = null,
  override val prescriberNihii: String? = null,
  override val relatedCode: String? = null,
  override val prescriptionDate: Long? = null,
  override val derogationMaxNumber: Int? = null,
  override val prescriberSsin: String? = null,
  override val prescriberLastName: String? = null,
  override val prescriberFirstName: String? = null,
  override val prescriberCdHcParty: String? = null,
  override val locationNihii: String? = null,
  override val locationCdHcParty: String? = null,
  override val locationService: Int? = null,
  override val admissionDate: Long? = null,
  override val canceled: Boolean? = null,
  override val accepted: Boolean? = null,
  override val pending: Boolean? = null,
  override val resent: Boolean? = null,
  override val archived: Boolean? = null,
  override val lost: Boolean? = null,
  override val insuranceJustification: Int? = null,
  override val cancelPatientInterventionReason: Int? = null,
  override val status: Long? = null,
  override val encryptedSelf: String? = null,
) : InvoicingCode

public fun InvoicingCode.copy(
  id: String? = this.id,
  dateCode: Long? = this.dateCode,
  logicalId: String? = this.logicalId,
  label: String? = this.label,
  userId: String? = this.userId,
  contactId: String? = this.contactId,
  serviceId: String? = this.serviceId,
  tarificationId: String? = this.tarificationId,
  code: String? = this.code,
  paymentType: PaymentType? = this.paymentType,
  paid: Double? = this.paid,
  totalAmount: Double? = this.totalAmount,
  reimbursement: Double? = this.reimbursement,
  patientIntervention: Double? = this.patientIntervention,
  doctorSupplement: Double? = this.doctorSupplement,
  conventionAmount: Double? = this.conventionAmount,
  vat: Double? = this.vat,
  error: String? = this.error,
  contract: String? = this.contract,
  contractDate: Long? = this.contractDate,
  units: Int? = this.units,
  side: Int? = this.side,
  timeOfDay: Int? = this.timeOfDay,
  eidReadingHour: Int? = this.eidReadingHour,
  eidReadingValue: String? = this.eidReadingValue,
  override3rdPayerCode: Int? = this.override3rdPayerCode,
  override3rdPayerReason: String? = this.override3rdPayerReason,
  transplantationCode: Int? = this.transplantationCode,
  prescriberNorm: Int? = this.prescriberNorm,
  percentNorm: Int? = this.percentNorm,
  prescriberNihii: String? = this.prescriberNihii,
  relatedCode: String? = this.relatedCode,
  prescriptionDate: Long? = this.prescriptionDate,
  derogationMaxNumber: Int? = this.derogationMaxNumber,
  prescriberSsin: String? = this.prescriberSsin,
  prescriberLastName: String? = this.prescriberLastName,
  prescriberFirstName: String? = this.prescriberFirstName,
  prescriberCdHcParty: String? = this.prescriberCdHcParty,
  locationNihii: String? = this.locationNihii,
  locationCdHcParty: String? = this.locationCdHcParty,
  locationService: Int? = this.locationService,
  admissionDate: Long? = this.admissionDate,
  canceled: Boolean? = this.canceled,
  accepted: Boolean? = this.accepted,
  pending: Boolean? = this.pending,
  resent: Boolean? = this.resent,
  archived: Boolean? = this.archived,
  lost: Boolean? = this.lost,
  insuranceJustification: Int? = this.insuranceJustification,
  cancelPatientInterventionReason: Int? = this.cancelPatientInterventionReason,
  status: Long? = this.status,
  encryptedSelf: String? = this.encryptedSelf,
): InvoicingCode {
                                      return when(this) {
                                          is DecryptedInvoicingCode -> copy(id = id, dateCode =
          dateCode, logicalId = logicalId, label = label, userId = userId, contactId = contactId,
          serviceId = serviceId, tarificationId = tarificationId, code = code, paymentType =
          paymentType, paid = paid, totalAmount = totalAmount, reimbursement = reimbursement,
          patientIntervention = patientIntervention, doctorSupplement = doctorSupplement,
          conventionAmount = conventionAmount, vat = vat, error = error, contract = contract,
          contractDate = contractDate, units = units, side = side, timeOfDay = timeOfDay,
          eidReadingHour = eidReadingHour, eidReadingValue = eidReadingValue, override3rdPayerCode =
          override3rdPayerCode, override3rdPayerReason = override3rdPayerReason, transplantationCode
          = transplantationCode, prescriberNorm = prescriberNorm, percentNorm = percentNorm,
          prescriberNihii = prescriberNihii, relatedCode = relatedCode, prescriptionDate =
          prescriptionDate, derogationMaxNumber = derogationMaxNumber, prescriberSsin =
          prescriberSsin, prescriberLastName = prescriberLastName, prescriberFirstName =
          prescriberFirstName, prescriberCdHcParty = prescriberCdHcParty, locationNihii =
          locationNihii, locationCdHcParty = locationCdHcParty, locationService = locationService,
          admissionDate = admissionDate, canceled = canceled, accepted = accepted, pending =
          pending, resent = resent, archived = archived, lost = lost, insuranceJustification =
          insuranceJustification, cancelPatientInterventionReason = cancelPatientInterventionReason,
          status = status, encryptedSelf = encryptedSelf)
      is EncryptedInvoicingCode -> copy(id = id, dateCode = dateCode, logicalId = logicalId, label =
          label, userId = userId, contactId = contactId, serviceId = serviceId, tarificationId =
          tarificationId, code = code, paymentType = paymentType, paid = paid, totalAmount =
          totalAmount, reimbursement = reimbursement, patientIntervention = patientIntervention,
          doctorSupplement = doctorSupplement, conventionAmount = conventionAmount, vat = vat, error
          = error, contract = contract, contractDate = contractDate, units = units, side = side,
          timeOfDay = timeOfDay, eidReadingHour = eidReadingHour, eidReadingValue = eidReadingValue,
          override3rdPayerCode = override3rdPayerCode, override3rdPayerReason =
          override3rdPayerReason, transplantationCode = transplantationCode, prescriberNorm =
          prescriberNorm, percentNorm = percentNorm, prescriberNihii = prescriberNihii, relatedCode
          = relatedCode, prescriptionDate = prescriptionDate, derogationMaxNumber =
          derogationMaxNumber, prescriberSsin = prescriberSsin, prescriberLastName =
          prescriberLastName, prescriberFirstName = prescriberFirstName, prescriberCdHcParty =
          prescriberCdHcParty, locationNihii = locationNihii, locationCdHcParty = locationCdHcParty,
          locationService = locationService, admissionDate = admissionDate, canceled = canceled,
          accepted = accepted, pending = pending, resent = resent, archived = archived, lost = lost,
          insuranceJustification = insuranceJustification, cancelPatientInterventionReason =
          cancelPatientInterventionReason, status = status, encryptedSelf = encryptedSelf)
                                          }
}

File: python-wrapper/src/python/model/embed/InvoicingCode.py
import json
from typing import Union, Dict, Optional
from model.embed.PaymentType import PaymentType
from model.specializations.Base64String import Base64String
from dataclasses import dataclass

InvoicingCode = Union['DecryptedInvoicingCode', 'EncryptedInvoicingCode']

def serialize_invoicing_code(invoicing_code: InvoicingCode) -> object:
	if isinstance(invoicing_code, DecryptedInvoicingCode):
		return {
			"type": "com.icure.sdk.model.embed.DecryptedInvoicingCode",
			"entity": invoicing_code.__serialize__()
		}
	elif isinstance(invoicing_code, EncryptedInvoicingCode):
		return {
			"type": "com.icure.sdk.model.embed.EncryptedInvoicingCode",
			"entity": invoicing_code.__serialize__()
		}
	else:
		raise Exception(f"{type(invoicing_code)} is not a known subclass of InvoicingCode")

def deserialize_invoicing_code(data: Union[str, Dict[str, object]]) -> 'InvoicingCode':
	deserialized_dict: dict[str, object]
	if isinstance(data, str):
		deserialized_dict = json.loads(data)
	else:
		deserialized_dict = data
	qualifier = deserialized_dict.get("type")
	if qualifier is None:
		raise Exception("Missing qualifier: type")
	if qualifier == "com.icure.sdk.model.embed.DecryptedInvoicingCode":
		DecryptedInvoicingCode._deserialize(deserialized_dict["entity"])
	elif qualifier == "com.icure.sdk.model.embed.EncryptedInvoicingCode":
		EncryptedInvoicingCode._deserialize(deserialized_dict["entity"])
	else:
		raise Exception(f"{qualifier} is not a known subclass of InvoicingCode")

@dataclass
class DecryptedInvoicingCode:
	id: Optional[str]
	date_code: Optional[int] = None
	logical_id: Optional[str] = None
	label: Optional[str] = None
	user_id: Optional[str] = None
	contact_id: Optional[str] = None
	service_id: Optional[str] = None
	tarification_id: Optional[str] = None
	code: Optional[str] = None
	payment_type: Optional['PaymentType'] = None
	paid: Optional[float] = None
	total_amount: Optional[float] = None
	reimbursement: Optional[float] = None
	patient_intervention: Optional[float] = None
	doctor_supplement: Optional[float] = None
	convention_amount: Optional[float] = None
	vat: Optional[float] = None
	error: Optional[str] = None
	contract: Optional[str] = None
	contract_date: Optional[int] = None
	units: Optional[int] = None
	side: Optional[int] = None
	time_of_day: Optional[int] = None
	eid_reading_hour: Optional[int] = None
	eid_reading_value: Optional[str] = None
	override3rd_payer_code: Optional[int] = None
	override3rd_payer_reason: Optional[str] = None
	transplantation_code: Optional[int] = None
	prescriber_norm: Optional[int] = None
	percent_norm: Optional[int] = None
	prescriber_nihii: Optional[str] = None
	related_code: Optional[str] = None
	prescription_date: Optional[int] = None
	derogation_max_number: Optional[int] = None
	prescriber_ssin: Optional[str] = None
	prescriber_last_name: Optional[str] = None
	prescriber_first_name: Optional[str] = None
	prescriber_cd_hc_party: Optional[str] = None
	location_nihii: Optional[str] = None
	location_cd_hc_party: Optional[str] = None
	location_service: Optional[int] = None
	admission_date: Optional[int] = None
	canceled: Optional[bool] = None
	accepted: Optional[bool] = None
	pending: Optional[bool] = None
	resent: Optional[bool] = None
	archived: Optional[bool] = None
	lost: Optional[bool] = None
	insurance_justification: Optional[int] = None
	cancel_patient_intervention_reason: Optional[int] = None
	status: Optional[int] = None
	encrypted_self: Optional['Base64String'] = None

	def __serialize__(self) -> object:
		return {
			"id": self.id,
			"dateCode": self.date_code,
			"logicalId": self.logical_id,
			"label": self.label,
			"userId": self.user_id,
			"contactId": self.contact_id,
			"serviceId": self.service_id,
			"tarificationId": self.tarification_id,
			"code": self.code,
			"paymentType": self.payment_type.__serialize__() if self.payment_type is not None else None,
			"paid": self.paid,
			"totalAmount": self.total_amount,
			"reimbursement": self.reimbursement,
			"patientIntervention": self.patient_intervention,
			"doctorSupplement": self.doctor_supplement,
			"conventionAmount": self.convention_amount,
			"vat": self.vat,
			"error": self.error,
			"contract": self.contract,
			"contractDate": self.contract_date,
			"units": self.units,
			"side": self.side,
			"timeOfDay": self.time_of_day,
			"eidReadingHour": self.eid_reading_hour,
			"eidReadingValue": self.eid_reading_value,
			"override3rdPayerCode": self.override3rd_payer_code,
			"override3rdPayerReason": self.override3rd_payer_reason,
			"transplantationCode": self.transplantation_code,
			"prescriberNorm": self.prescriber_norm,
			"percentNorm": self.percent_norm,
			"prescriberNihii": self.prescriber_nihii,
			"relatedCode": self.related_code,
			"prescriptionDate": self.prescription_date,
			"derogationMaxNumber": self.derogation_max_number,
			"prescriberSsin": self.prescriber_ssin,
			"prescriberLastName": self.prescriber_last_name,
			"prescriberFirstName": self.prescriber_first_name,
			"prescriberCdHcParty": self.prescriber_cd_hc_party,
			"locationNihii": self.location_nihii,
			"locationCdHcParty": self.location_cd_hc_party,
			"locationService": self.location_service,
			"admissionDate": self.admission_date,
			"canceled": self.canceled,
			"accepted": self.accepted,
			"pending": self.pending,
			"resent": self.resent,
			"archived": self.archived,
			"lost": self.lost,
			"insuranceJustification": self.insurance_justification,
			"cancelPatientInterventionReason": self.cancel_patient_intervention_reason,
			"status": self.status,
			"encryptedSelf": self.encrypted_self,
		}

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'DecryptedInvoicingCode':
		deserialized_dict: dict[str, object]
		if isinstance(data, str):
			deserialized_dict = json.loads(data)
		else:
			deserialized_dict = data
		return cls(
			id = deserialized_dict.get("id"),
			date_code = deserialized_dict.get("dateCode"),
			logical_id = deserialized_dict.get("logicalId"),
			label = deserialized_dict.get("label"),
			user_id = deserialized_dict.get("userId"),
			contact_id = deserialized_dict.get("contactId"),
			service_id = deserialized_dict.get("serviceId"),
			tarification_id = deserialized_dict.get("tarificationId"),
			code = deserialized_dict.get("code"),
			payment_type = PaymentType._deserialize(deserialized_dict.get("paymentType")) if deserialized_dict.get("paymentType") is not None else None,
			paid = deserialized_dict.get("paid"),
			total_amount = deserialized_dict.get("totalAmount"),
			reimbursement = deserialized_dict.get("reimbursement"),
			patient_intervention = deserialized_dict.get("patientIntervention"),
			doctor_supplement = deserialized_dict.get("doctorSupplement"),
			convention_amount = deserialized_dict.get("conventionAmount"),
			vat = deserialized_dict.get("vat"),
			error = deserialized_dict.get("error"),
			contract = deserialized_dict.get("contract"),
			contract_date = deserialized_dict.get("contractDate"),
			units = deserialized_dict.get("units"),
			side = deserialized_dict.get("side"),
			time_of_day = deserialized_dict.get("timeOfDay"),
			eid_reading_hour = deserialized_dict.get("eidReadingHour"),
			eid_reading_value = deserialized_dict.get("eidReadingValue"),
			override3rd_payer_code = deserialized_dict.get("override3rdPayerCode"),
			override3rd_payer_reason = deserialized_dict.get("override3rdPayerReason"),
			transplantation_code = deserialized_dict.get("transplantationCode"),
			prescriber_norm = deserialized_dict.get("prescriberNorm"),
			percent_norm = deserialized_dict.get("percentNorm"),
			prescriber_nihii = deserialized_dict.get("prescriberNihii"),
			related_code = deserialized_dict.get("relatedCode"),
			prescription_date = deserialized_dict.get("prescriptionDate"),
			derogation_max_number = deserialized_dict.get("derogationMaxNumber"),
			prescriber_ssin = deserialized_dict.get("prescriberSsin"),
			prescriber_last_name = deserialized_dict.get("prescriberLastName"),
			prescriber_first_name = deserialized_dict.get("prescriberFirstName"),
			prescriber_cd_hc_party = deserialized_dict.get("prescriberCdHcParty"),
			location_nihii = deserialized_dict.get("locationNihii"),
			location_cd_hc_party = deserialized_dict.get("locationCdHcParty"),
			location_service = deserialized_dict.get("locationService"),
			admission_date = deserialized_dict.get("admissionDate"),
			canceled = deserialized_dict.get("canceled"),
			accepted = deserialized_dict.get("accepted"),
			pending = deserialized_dict.get("pending"),
			resent = deserialized_dict.get("resent"),
			archived = deserialized_dict.get("archived"),
			lost = deserialized_dict.get("lost"),
			insurance_justification = deserialized_dict.get("insuranceJustification"),
			cancel_patient_intervention_reason = deserialized_dict.get("cancelPatientInterventionReason"),
			status = deserialized_dict.get("status"),
			encrypted_self = deserialized_dict.get("encryptedSelf"),
		)

@dataclass
class EncryptedInvoicingCode:
	id: Optional[str]
	date_code: Optional[int] = None
	logical_id: Optional[str] = None
	label: Optional[str] = None
	user_id: Optional[str] = None
	contact_id: Optional[str] = None
	service_id: Optional[str] = None
	tarification_id: Optional[str] = None
	code: Optional[str] = None
	payment_type: Optional['PaymentType'] = None
	paid: Optional[float] = None
	total_amount: Optional[float] = None
	reimbursement: Optional[float] = None
	patient_intervention: Optional[float] = None
	doctor_supplement: Optional[float] = None
	convention_amount: Optional[float] = None
	vat: Optional[float] = None
	error: Optional[str] = None
	contract: Optional[str] = None
	contract_date: Optional[int] = None
	units: Optional[int] = None
	side: Optional[int] = None
	time_of_day: Optional[int] = None
	eid_reading_hour: Optional[int] = None
	eid_reading_value: Optional[str] = None
	override3rd_payer_code: Optional[int] = None
	override3rd_payer_reason: Optional[str] = None
	transplantation_code: Optional[int] = None
	prescriber_norm: Optional[int] = None
	percent_norm: Optional[int] = None
	prescriber_nihii: Optional[str] = None
	related_code: Optional[str] = None
	prescription_date: Optional[int] = None
	derogation_max_number: Optional[int] = None
	prescriber_ssin: Optional[str] = None
	prescriber_last_name: Optional[str] = None
	prescriber_first_name: Optional[str] = None
	prescriber_cd_hc_party: Optional[str] = None
	location_nihii: Optional[str] = None
	location_cd_hc_party: Optional[str] = None
	location_service: Optional[int] = None
	admission_date: Optional[int] = None
	canceled: Optional[bool] = None
	accepted: Optional[bool] = None
	pending: Optional[bool] = None
	resent: Optional[bool] = None
	archived: Optional[bool] = None
	lost: Optional[bool] = None
	insurance_justification: Optional[int] = None
	cancel_patient_intervention_reason: Optional[int] = None
	status: Optional[int] = None
	encrypted_self: Optional['Base64String'] = None

	def __serialize__(self) -> object:
		return {
			"id": self.id,
			"dateCode": self.date_code,
			"logicalId": self.logical_id,
			"label": self.label,
			"userId": self.user_id,
			"contactId": self.contact_id,
			"serviceId": self.service_id,
			"tarificationId": self.tarification_id,
			"code": self.code,
			"paymentType": self.payment_type.__serialize__() if self.payment_type is not None else None,
			"paid": self.paid,
			"totalAmount": self.total_amount,
			"reimbursement": self.reimbursement,
			"patientIntervention": self.patient_intervention,
			"doctorSupplement": self.doctor_supplement,
			"conventionAmount": self.convention_amount,
			"vat": self.vat,
			"error": self.error,
			"contract": self.contract,
			"contractDate": self.contract_date,
			"units": self.units,
			"side": self.side,
			"timeOfDay": self.time_of_day,
			"eidReadingHour": self.eid_reading_hour,
			"eidReadingValue": self.eid_reading_value,
			"override3rdPayerCode": self.override3rd_payer_code,
			"override3rdPayerReason": self.override3rd_payer_reason,
			"transplantationCode": self.transplantation_code,
			"prescriberNorm": self.prescriber_norm,
			"percentNorm": self.percent_norm,
			"prescriberNihii": self.prescriber_nihii,
			"relatedCode": self.related_code,
			"prescriptionDate": self.prescription_date,
			"derogationMaxNumber": self.derogation_max_number,
			"prescriberSsin": self.prescriber_ssin,
			"prescriberLastName": self.prescriber_last_name,
			"prescriberFirstName": self.prescriber_first_name,
			"prescriberCdHcParty": self.prescriber_cd_hc_party,
			"locationNihii": self.location_nihii,
			"locationCdHcParty": self.location_cd_hc_party,
			"locationService": self.location_service,
			"admissionDate": self.admission_date,
			"canceled": self.canceled,
			"accepted": self.accepted,
			"pending": self.pending,
			"resent": self.resent,
			"archived": self.archived,
			"lost": self.lost,
			"insuranceJustification": self.insurance_justification,
			"cancelPatientInterventionReason": self.cancel_patient_intervention_reason,
			"status": self.status,
			"encryptedSelf": self.encrypted_self,
		}

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'EncryptedInvoicingCode':
		deserialized_dict: dict[str, object]
		if isinstance(data, str):
			deserialized_dict = json.loads(data)
		else:
			deserialized_dict = data
		return cls(
			id = deserialized_dict.get("id"),
			date_code = deserialized_dict.get("dateCode"),
			logical_id = deserialized_dict.get("logicalId"),
			label = deserialized_dict.get("label"),
			user_id = deserialized_dict.get("userId"),
			contact_id = deserialized_dict.get("contactId"),
			service_id = deserialized_dict.get("serviceId"),
			tarification_id = deserialized_dict.get("tarificationId"),
			code = deserialized_dict.get("code"),
			payment_type = PaymentType._deserialize(deserialized_dict.get("paymentType")) if deserialized_dict.get("paymentType") is not None else None,
			paid = deserialized_dict.get("paid"),
			total_amount = deserialized_dict.get("totalAmount"),
			reimbursement = deserialized_dict.get("reimbursement"),
			patient_intervention = deserialized_dict.get("patientIntervention"),
			doctor_supplement = deserialized_dict.get("doctorSupplement"),
			convention_amount = deserialized_dict.get("conventionAmount"),
			vat = deserialized_dict.get("vat"),
			error = deserialized_dict.get("error"),
			contract = deserialized_dict.get("contract"),
			contract_date = deserialized_dict.get("contractDate"),
			units = deserialized_dict.get("units"),
			side = deserialized_dict.get("side"),
			time_of_day = deserialized_dict.get("timeOfDay"),
			eid_reading_hour = deserialized_dict.get("eidReadingHour"),
			eid_reading_value = deserialized_dict.get("eidReadingValue"),
			override3rd_payer_code = deserialized_dict.get("override3rdPayerCode"),
			override3rd_payer_reason = deserialized_dict.get("override3rdPayerReason"),
			transplantation_code = deserialized_dict.get("transplantationCode"),
			prescriber_norm = deserialized_dict.get("prescriberNorm"),
			percent_norm = deserialized_dict.get("percentNorm"),
			prescriber_nihii = deserialized_dict.get("prescriberNihii"),
			related_code = deserialized_dict.get("relatedCode"),
			prescription_date = deserialized_dict.get("prescriptionDate"),
			derogation_max_number = deserialized_dict.get("derogationMaxNumber"),
			prescriber_ssin = deserialized_dict.get("prescriberSsin"),
			prescriber_last_name = deserialized_dict.get("prescriberLastName"),
			prescriber_first_name = deserialized_dict.get("prescriberFirstName"),
			prescriber_cd_hc_party = deserialized_dict.get("prescriberCdHcParty"),
			location_nihii = deserialized_dict.get("locationNihii"),
			location_cd_hc_party = deserialized_dict.get("locationCdHcParty"),
			location_service = deserialized_dict.get("locationService"),
			admission_date = deserialized_dict.get("admissionDate"),
			canceled = deserialized_dict.get("canceled"),
			accepted = deserialized_dict.get("accepted"),
			pending = deserialized_dict.get("pending"),
			resent = deserialized_dict.get("resent"),
			archived = deserialized_dict.get("archived"),
			lost = deserialized_dict.get("lost"),
			insurance_justification = deserialized_dict.get("insuranceJustification"),
			cancel_patient_intervention_reason = deserialized_dict.get("cancelPatientInterventionReason"),
			status = deserialized_dict.get("status"),
			encrypted_self = deserialized_dict.get("encryptedSelf"),
		)

__all__ = ['InvoicingCode','serialize_invoicing_code','deserialize_invoicing_code','DecryptedInvoicingCode','EncryptedInvoicingCode']